### PR TITLE
feature/BI-12229-Implementing-DownloadFile

### PR DESCRIPTION
**Pull request by OsvaldoMartini**
Implementing Download the File from S3
and Clean up the code

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,9 +35,11 @@
 		<api-sdk-manager-java-library.version>1.0.4</api-sdk-manager-java-library.version>
 		<api-sdk-java.version>4.3.24</api-sdk-java.version>
 		<api-helper-java-library.version>1.4.5</api-helper-java-library.version>
-		<jaxb-api.version>2.3.1</jaxb-api.version>
 		<aws-java-sdk.version>1.12.343</aws-java-sdk.version>
 		<api-helper-java.version>1.4.3</api-helper-java.version>
+
+		<!-- java apis libs -->
+		<jaxb-api.version>2.3.1</jaxb-api.version>
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply pom.xml:42]: this Class use it
package uk.gov.companieshouse.account.validator.validation.ixbrl;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(name = "data")
public class Data {

File: src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
-
-            // TODO
-            // IF ZIP FILE calls tnedp valildation No ixbrl_image_renderer
-            // IF NO ZIP FILE calls ixbrl/felix valildation call to ixbrl_image_renderer
-
-            message = "Uploaded the file successfully";
-            return ResponseEntity.status(HttpStatus.OK).body(new ResponseMessage(message));
+            boolean result = accountValidatorImpl.downloadIxbrlFromLocation(fileDetails);
+            if (result) {
+                message = "File validated successfully";
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java:38]: I agree
Shall we use the same response results using FelixValidor showing the error tags, and any errors found? like i.e: BalanceSheetDate, companiesHouseRegisteredNumber, errorMessages
or should we respond Passed or Failed?

File: src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java [hunk 3]
```diff
@@ -1,41 +1,48 @@
...
+                return ResponseEntity.status(HttpStatus.OK).body(new ResponseMessage(message));
+            } else {
+                message = "File validation failed";
+                return ResponseEntity.status(HttpStatus.EXPECTATION_FAILED).body(new ResponseMessage(message));
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java:42]: Will do the changes for 200

File: src/main/java/uk/gov/companieshouse/account/validator/service/impl/FelixValidationServiceImpl.java [hunk 4]
```diff
@@ -147,13 +146,13 @@ private LinkedMultiValueMap<String, Object> createFileMessageResource(String ixb
...
     /**
-     * Obtain the URL of the TNEP validator from the environment
+     * Obtain the URL of the FELIX validator from the environment
      *
      * @return String
      */
     protected String getIxbrlValidatorUri() {
 
-        return environmentReader.getMandatoryString(IXBRL_VALIDATOR_URI);
+        return environmentReader.getMandatoryString(FELIX_VALIDATOR_URI);
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/java/uk/gov/companieshouse/account/validator/service/impl/FelixValidationServiceImpl.java:155]: agree doing the changes

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -28,5 +28,4 @@ info.java-vendor = ${java.specification.vendor}
 management.endpoints.web.exposure.include=health,mappings
 management.endpoint.shutdown.enabled=true
 
-
-ACCOUNTS_API_PORT=18624
\ No newline at end of file
+uk.gov.ch.account.validator.platformMaxDecodedSizeMB=300
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/resources/application.properties:31]: agree

File: src/main/java/uk/gov/companieshouse/account/validator/service/impl/AccountValidatorImpl.java [hunk 6]
```diff
@@ -2,70 +2,147 @@
...
     private static final String LOG_MESSAGE_KEY = "message";
-    private final FileTransferTool fileTransferTool;
-    private final TnepValidationService tnepValidationService;
+    private static final String LOG_ERROR_KEY = "error";
+    private static final String LOCATION = "s3://%s/%s/%s";
+    private static final String DOCUMENT_BUCKET_NAME = "DOCUMENT_BUCKET_NAME";
+    private static final String DOCUMENT_RENDER_SERVICE_HOST = "DOCUMENT_RENDER_SERVICE_HOST";
+    private String documentBucketName;
+
+    @Autowired
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/java/uk/gov/companieshouse/account/validator/service/impl/AccountValidatorImpl.java:32]: modified. 
Done

File: src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java [hunk 7]
```diff
@@ -1,41 +1,48 @@
...
+            }
         } catch (Exception e) {
-            message = "Could not upload the file: " + fileDetails.getFile_name()+ ". Error: " + e.getMessage();
+            message = "Could not validate the file: " + fileDetails.getFile_name() + ". Error: " + e.getMessage();
             return ResponseEntity.status(HttpStatus.EXPECTATION_FAILED).body(new ResponseMessage(message));
```
[the remark this answers is not in the document]
Comment (OsvaldoMartini) [reply src/main/java/uk/gov/companieshouse/account/validator/controller/FilesController.java]: Request response from account-api to account-web and logs to be defined. 
Shall we use the same responses from Felix/ixbrl objects?

**Comment by OsvaldoMartini** [comment]
Need to Clean Up and work

**Review by HarryBosworth — APPROVED** [review]

**Pull request merged**